test/mocks/server/transport_socket_factory_context.h

```diff
@@ -32,7 +32,7 @@ class MockTransportSocketFactoryContext : public TransportSocketFactoryContext {
   MOCK_METHOD(Stats::Scope&, statsScope, ());
   MOCK_METHOD(Init::Manager&, initManager, ());
 
-  testing::NiceMock<StatelessMockServerFactoryContext> server_context_;
+  testing::NiceMock<MockServerFactoryContext> server_context_;
```

Comment:
@wbpcode If I dont do this change, `examples_config_test` fails when it tests the example config that is being added. It fails in initializing `GenericSecretSdsApi`:
```
std::make_shared<GenericSecretSdsApi>(
        sds_config, sds_config_name, secret_provider_context.clusterManager().subscriptionFactory(),
        server_context.mainThreadDispatcher().timeSource(),
        secret_provider_context.messageValidationVisitor(), server_context.serverScope().store(),
        destructor_cb, server_context.mainThreadDispatcher(), server_context.api());
```

Not sure if it will bomb in other asan, tsan tests.

Reply:
asan/tsan tests failing because of using `MockServerFactoryContext`